pkg/controller/node/node_controller.go

```diff
@@ -976,22 +980,32 @@ func (ctrl *Controller) updateCandidateMachines(pool *mcfgv1.MachineConfigPool,
 	targetConfig := pool.Spec.Configuration.Name
 	for _, node := range candidates {
 
-		// Check image details for this pool
-		targetImage, equivalentTo, isImagePool, targetImageMatchesConfig := ctrl.experimentalHasValidImage(pool)
-
-		// If our pool is annotated with an image AND that image is the right image, then update these nodes with it
-		// TODO(jkyros): this does not obviate the desiredConfig annotation below for now, as we're using "are we in our desired machineconfig"
-		// as our "done" signal
-		if isImagePool {
-			if targetImageMatchesConfig {
-				ctrl.logPool(pool, "Setting node %s target image to %s", node.Name, targetImage)
-				if err := ctrl.setNodeAnnotation(node.Name, daemonconsts.DesiredImageConfigAnnotationKey, targetImage); err != nil {
-					return goerrs.Wrapf(err, "setting desired config for node %s", node.Name)
+		// double check that this node has CoreOS and will support layering
+		osID, ok := node.Labels[osIDLabel]
+		if ok {
+			if daemon.IsProbablyCoreOSVariant(osID) {
```

Comment:
Is it okay for me to use daemon code in the node_controller?

Reply:
I don't see the problem with that, though I guess it looks like `ctrlcommon` is a bit cleaner.